primo/opt_model/model_with_clustering.py

```diff
@@ -353,6 +377,9 @@ def __init__(self, model_inputs, *args, **kwargs):
         if model_inputs.config.max_wells_per_owner is not None:
             self.add_owner_well_count()
 
+        if budget_sufficient is False:
```

Comment:
I think this would be good to have as something that is completely up to the user to add to the model. The user should be able to tell the model whether or not to use the slack variable with the budget constraint, and then if they elect to do so, then the budget_sufficient logic would apply.